### ISSUE TITLE
Consider dropping explicit pinning on anaconda metapackage, letting versions float up

### ISSUE THREAD

**Issue by kalefranz** [ContinuumIO/anaconda-issues#4298]
I'd like to document for discussion a proposal to change the nature of the anaconda metapackage.  Right now, all versions of dependencies within the package are explicitly pinned.  This is because Continuum provides extra levels of QA and testing guarantees that all of the packages contained within are known to be compatible and working together.

I think a lot (many? most?) or our users have a different expectation of the anaconda metapackage.  I think they conceive of anaconda as just a minimal set of software and dependencies that includes python.  They're not enterprise customers who are interested in strict stability, one of the cases where explicit pinning makes the most sense.

My proposal is that, perhaps with a major version bump (which is appropriate for a change of this magnitude), we

1. list in the anaconda metapackage only the "top level" packages we actually consider part of Anaconda, and
2. set each of those versions as `>=`, rather than explicit and strict `==` pinning (with the exception of Anaconda2, where we'd need `python >=2.7.12,<3`)

In parallel, we can create and release an `anaconda-pinned` metapackage, which has the strict pinning of the current scheme.

I really believe this is more in line with what most users expect.  It obviates the need for the hacky `anaconda-custom` that leads to a lot of confusion.  In fact, comparing download numbers of the `anaconda-custom` package to download numbers for the Anaconda installer would be a good measure of user behavior here.

This proposal would also make `conda update --all` much more friendly.  We wouldn't have to jump through hoops to tell people they have to `conda remove anaconda` before they `conda update --all` or write seemingly complicated explanations like https://stackoverflow.com/questions/45197777/how-do-i-update-anaconda for something that could/should really be pretty simple.

**Comment by ijstokes** [comment]
FWIW, I agree completely with this assessment.  I've had to go and create [things like this](https://gist.github.com/ijstokes/45605149213e1630e928d8ff4cbdbe5f) to get around this problem (effectively my own version of a versionless anaconda meta-package dependency list).

**Comment by mingwandroid** [comment]
There's nothing minimal about anaconda. IMHO it contains loads of stuff the average user will never ever run.

I know this is bikeshedding, but I would call the 'baked' packages `anaconda-release` instead. Pinned is, more or less an expert-level term and somewhat Continuum-y.

Wouldn't you need to remove the `anaconda` package to be able to install an `anaconda-release` package though (to prevent conflicts when packages get deprecated)?

**Comment by kalefranz** [comment]
I agree that the "pinned" version should be called `anaconda-release`.  That's a better name.

**Comment by kalefranz** [comment]
https://www.anaconda.com/blog/developer-blog/keeping-anaconda-date/

**Comment by kalefranz** [comment]
https://groups.google.com/a/continuum.io/forum/#!msg/conda/OQm8wHfO5SE/joqQovZdCQAJ

**Comment by lucatrv** [comment]
I propose a slightly different approach. The "anaconda" metapackage could still be "pinned" to specific package versions, to ensure that at any time all installed packages are mutually compatible. However the "conda update --all" could always update it to the last available version, even in between official releases (similarly to nowadays "custom" versions). This way, the "anaconda" metapackage would be periodically updated (without waiting official releases), together with all dependent packages. This would be the default behavior for most users.

Then there could be an "anaconda-release" metapackage similar to the "anaconda" one (same required packages) but it would never be updated by the "conda update --all" command (similarly to the current behavior). This way, if someone wanted to keep a fixed released environment (e.g. enterprise customers),
he would just need to replace the "anaconda" with the "anaconda-release" metapackage. The "anaconda-release" metapackage would never be updated in between official releases, and would require manual update between official releases, with the command "conda update anaconda-release" (this would pull updates only four times a year).

**Comment by lucatrv** [comment]
Here's another slightly different proposal. To make two metapackages available: one named "anaconda-rolling" which would be regularly updated (similarly to today's "custom" metapackage), and the other one named "anaconda-release" to be updated only four times a year (with major Anaconda releases). During the first installation, the user would be asked whether to install the "rolling" or the "release" metapackage. The "conda update --all" would update either one. At any time the user would be able to pass from one updating behavior to the other one simply replacing the "anaconda" metapackage.
